### ISSUE TITLE
Upload: file validation with Symfony\Validator?

### ISSUE THREAD

**Issue by Gounlaf** [riotkit-org/file-repository#18]
Since Silex is Symfony, should we use https://silex.sensiolabs.org/doc/2.0/providers/validator.html ?
There are a lot of built-in validator, like validation of "uploaded files" (for `UploadByHttpActionHandler::handleValidation()`)...

I don't know (yet) if there is a lot of validation elsewhere in the application.

Pros/Cons is like discussing about reinvent the wheel. I understand that you want the application to be the smaller as possible. IMHO, very common - and hard, that need to be tested and retested again - things should rely on "common packages".

**Comment by blackandred** [comment]
Hello,

I'm against reinventing the wheel over and over again.
So, I think that it is a very good idea to use the `Symfony\Validator` and optionally `Symfony\Form` if it would be handy.

Generally now my point of view has changed and I think that selecting Silex was wrong and I would select only Symfony for all projects. So, using a common Symfony component from my opinion looks very good.

Thanks! :+1:

**Comment by Gounlaf** [comment]
> Generally now my point of view has changed and I think that selecting Silex was wrong and I would select only Symfony for all projects.

Silex is like Lumen for Laravel :) They are for fast "simple" project. A "micro-service" is a simple project :)

**Comment by blackandred** [comment]
I noticed that using a full framework is in most cases more practical even for microservices. Really, at commercial work my team have too many technical failures on production because of using Lumen instead of fully featured framework. Its stripped out of everything, especially no configuration validation, no environment separation.

**Comment by blackandred** [comment]
I guess that the `file-repository` will become a service soon, or it could be already named a `service` :wink: 
I think that is ok, its still working very well on a cheap shared hosting, and that was the original target to achieve.

**Comment by Gounlaf** [comment]
I found your project after long hours of research.

I'm building an API (_API-Centric_ should rules the world), based on [api-platform](https://github.com/api-platform/api-platform), and "file uploading" was the missing feature. I look over and over a lot of "Build your API" project (php or nodejs), and unfortunately, there were always something missing:
* OAuth2 (with non-deprecated packages)
* File uploading

I found somewhere (don't remember) a good suggestions: build your API somewhere, and store your files with something else.
That's how I finally found this project. It does only what's it's expected, and that's good!

**Comment by blackandred** [comment]
Indeed. That's the rule how this project was created.
I started creating a portal and had a lot of things to do, to allow uploading files (with images cropping), collecting events from facebook (had to write a [microservice that would proxy the traffic](https://github.com/Wolnosciowiec/web-proxy) more smarter than other already existing projects), implement a hotlinking mechanism to serve hotlinked resources from eg. facebook as long as possible then switch to the `file-repository` - there is [other micro service](https://github.com/Wolnosciowiec/remote-resource-verifier) that is verifying remote resources. 

Finally the microservice infrastructure started to be hard to debug, so I implemented a microservice for logging purposes (as my non-commerical project does not have possibility to buy a New Relic for $5000 month), so [the notification project was born](https://github.com/Wolnosciowiec/wolnosciowiec-notification).
